solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -78,6 +81,39 @@
   static final int MOD_NOT     = 10;
   static final int MOD_REQ     = 11;
 
+  protected SynonymQueryStyle synonymQueryStyle = AS_SAME_TERM;
+
+  /**
+   *  Query strategy when analyzed query terms overlap the same position (ie synonyms)
+   *  consider if pants and khakis are query time synonyms
+   *
+   *  <li>{@link #AS_SAME_TERM}</li>
+   *  <li>{@link #PICK_BEST}</li>
+   *  <li>{@link #AS_DISTINCT_TERMS}</li>
+   */
+  public static enum SynonymQueryStyle {
```

Comment:
I like the new name, and thanks for improving the javadocs.  BTW that "li" HTML list is missing the "<ul> wrapper.  Or better IMO is simply drop this list; it has no value I think.